### PR TITLE
EAS-2209 Adding Dominators

### PR DESCRIPTION
**Pull request by jxuatlas**

### CODE REVIEW

File: crates/forge_analyzer/src/ir.rs [hunk 1]
```diff
@@ -392,13 +416,185 @@ impl Body {
...
+    pub(crate) fn dominance_frontier(&self, b: BasicBlockId) -> impl Iterator<Item = BasicBlockId> + '_ {
+        let idom = &self.dominator_tree().idom;
+        let mut frontiers: Vec<Vec<BasicBlockId>> = Vec::new();
+        for _ in 0..self.blocks.len() {
+            frontiers.push(Vec::new());
+        }
+
+        // algorithm from https://en.wikipedia.org/wiki/Static_single-assignment_form#Computing_minimal_SSA_using_dominance_frontiers
+        // which references: https://www.cs.tufts.edu/comp/150FP/archive/keith-cooper/dom14.pdf
+        for (id, block) in self.iter_blocks_enumerated() {
```
Comment (aliner-wang) [inline crates/forge_analyzer/src/ir.rs:567]: replace "block" with _ since you're not using "block" in the loop.

File: crates/fsrt/src/main.rs [hunk 2]
```diff
@@ -171,6 +175,12 @@ pub(crate) fn scan_directory<'a>(
         std::process::exit(0);
     }
 
+    if let Some(func) = opts.dump_dt.as_ref() {
+        println!("My CLI thing WORKS");
```
Comment (aliner-wang) [inline crates/fsrt/src/main.rs:179]: This was mb. Can you delete this line? ty <3

File: crates/forge_analyzer/src/ir.rs (hunk 1 above)
Comment (jxuatlas) [reply crates/forge_analyzer/src/ir.rs:567]: noted - thank you @aliner-wang! I think I have some other places with the same issue, will make those changes as well

File: crates/fsrt/src/main.rs [hunk 3]
```diff
@@ -343,6 +353,7 @@ pub(crate) fn scan_directory<'a>(
 
 fn main() -> Result<()> {
     let mut args = Args::parse();
+    println!("What is args {:?}", args);
```
Comment (aliner-wang) [inline crates/fsrt/src/main.rs:356]: Also delete this line pls or keep it (up to you) <3

**Pull request merged**